python/mrc/_pymrc/src/module_definitions/mirror_tap_util.cpp

```diff
@@ -0,0 +1,117 @@
+#include "mrc/segment/builder.hpp"
+#include "mrc/version.hpp"
+
+#include <nlohmann/json.hpp>
+#include <pybind11/functional.h>
+#include <pybind11/pybind11.h>
+#include <pybind11/pytypes.h>
+
+#include <string>
+
+const std::vector<unsigned int> PybindSegmentModuleVersion{mrc_VERSION_MAJOR, mrc_VERSION_MINOR, mrc_VERSION_PATCH};
+
+namespace py = pybind11;
+
+namespace {
+class MirrorTapUtilProxy
+{
+    using py_mirror_tap_t = mrc::modules::MirrorTapUtil<mrc::pymrc::PyHolder>;
+
+  public:
+    static std::shared_ptr<py_mirror_tap_t> create(const std::string& name)
+    {
+        return std::make_shared<py_mirror_tap_t>(name);
+    }
+
+    static std::shared_ptr<py_mirror_tap_t> create(const std::string& name, py::dict config)
+    {
+        return std::make_shared<py_mirror_tap_t>(name, mrc::pymrc::cast_from_pyobject(config));
+    }
+
+    static py::list create_or_extend_ingress_ports(py_mirror_tap_t& self, py::list ingress_ports)
+    {
+        ingress_ports.append(self.get_ingress_tap_name());
+        return ingress_ports;
+    }
+
+    static py::list create_or_extend_egress_ports(py_mirror_tap_t& self, py::list ingress_ports)
+    {
+        ingress_ports.append(self.get_ingress_tap_name());
+        return ingress_ports;
+    }
+};
+}  // namespace
+
+namespace mrc::pymrc {
+[[maybe_unused]] void register_mirror_tap_modules()
+{
+    using namespace mrc::modules;
+
+    ModelRegistryUtil::create_registered_module<mrc::modules::MirrorTapModule<pymrc::PyHolder>>(
+        "MirrorTap",
+        "mrc",
+        PybindSegmentModuleVersion);
+
+    ModelRegistryUtil::create_registered_module<MirrorTapStreamModule<pymrc::PyHolder>>("MirrorStreamBufferImmediate",
+                                                                                        "mrc",
+                                                                                        PybindSegmentModuleVersion);
+
+    ModelRegistryUtil::create_registered_module<StreamBufferModule<pymrc::PyHolder>>("StreamBufferImmediate",
+                                                                                     "mrc",
+                                                                                     PybindSegmentModuleVersion);
+}
+
+[[maybe_unused]] void init_mirror_tap_util(py::module_& smodule)
+{
+    using python_mirror_tap_util_t = mrc::modules::MirrorTapUtil<pymrc::PyHolder>;
+    auto MirrorTap = py::class_<python_mirror_tap_util_t, std::shared_ptr<python_mirror_tap_util_t>>(smodule,
+                                                                                                     "MirrorTap");
+
+    MirrorTap.def(py::init(py::overload_cast<const std::string&>(&::MirrorTapUtilProxy::create)),
+                  py::return_value_policy::take_ownership);
+
+    MirrorTap.def(py::init(py::overload_cast<const std::string&, py::dict>(&::MirrorTapUtilProxy::create)),
+                  py::return_value_policy::take_ownership);
```

Comment:
As we discussed, I would prefer that the python module initialization code remain in the `python/mrc/core` where the modules will be created. All files under `python/mrc/_pymrc` are for creating shared utility classes used by the python modules.